public/app/percona/shared/components/Elements/PlatformConnectedLoader/PlatformConnectedLoader.tsx

```diff
@@ -5,23 +5,36 @@ import { Messages } from './PlatformConnectedLoader.messages';
 import { PLATFORM_SETTINGS_URL } from './PlatformConnectedLoader.constants';
 import { getStyles } from './PlatformConnectedLoader.styles';
 import { PermissionLoader } from '../PermissionLoader';
+import { useSelector } from 'react-redux';
+import { getPerconaUser } from 'app/percona/shared/core/selectors';
+import { EmptyBlock } from '../EmptyBlock';
 
 export const PlatformConnectedLoader: FC = ({ children }) => {
   const styles = useStyles(getStyles);
-  const featureSelector = useCallback((state: StoreState) => !!state.perconaUser.isPlatformUser, []);
+  const featureSelector = useCallback((state: StoreState) => !!state.perconaSettings.isConnectedToPortal, []);
+  const { isPlatformUser } = useSelector(getPerconaUser);
+
+  const checkForPlatformUser = () => {
+    if (isPlatformUser) {
+      return children;
+    }
+    return <EmptyBlock dataTestId="not-platform-user">{Messages.platformUser}</EmptyBlock>;
+  };
+
+  const ErrorMessage = () => (
```

Comment:
I think we should declare this component outside of here, as a component. Not necessarily a new file, but just declare it outside.

Reply:
I moved it to separate file to make it more consistent